From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jagan Teki <jagan@amarulasolutions.com>
Date: Wed, 16 Feb 2022 16:16:07 +0530
Subject: [PATCH] arm64: dts: rk3399-rock-pi-4c: Enable Display Port

Enable Display Port on ROCK Pi 4C board.

Unlike, other RK3399 platforms with accessing DP Altmode
via Type-C connector, the display Port on ROCK Pi 4C is
accessible via physical display port connector by means
of Type-C Virtual PD extcon configuration.

Enable support for it.

Signed-off-by: Jagan Teki <jagan@amarulasolutions.com>
---
 .../boot/dts/rockchip/rk3399-rock-pi-4c.dts   | 42 +++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts
index 8e9ac6bc2cde..9e6e1c7e7d86 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts
@@ -15,6 +15,31 @@ / {
 	aliases {
 		mmc2 = &sdio0;
 	};
+
+	vcc3v3_dp: vcc3v3-dp-regulator {
+		compatible = "regulator-fixed";
+		enable-active-high;
+		gpio = <&gpio1 RK_PD0 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&dp_pwr_en>;
+		regulator-name = "vcc3v3_dp";
+		regulator-always-on;
+		regulator-boot-on;
+		vin-supply = <&vcc3v3_sys>;
+	};
+
+	virtual_pd: virtual-pd {
+		compatible = "linux,extcon-pd-virtual";
+		hpd-gpios = <&gpio4 25 GPIO_ACTIVE_LOW>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&hpd_en>;
+		dp-pwr-supply = <&vcc3v3_dp>;
+	};
+};
+
+&cdn_dp {
+	extcon = <&virtual_pd>;
+	status = "okay";
 };
 
 &es8316 {
@@ -30,6 +55,18 @@ vcc5v0_host_en: vcc5v0-host-en {
 			rockchip,pins = <3 RK_PD6 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	hpd {
+		hpd_en: hpd-en {
+			rockchip,pins = <4 RK_PD1 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+	dp {
+		dp_pwr_en: dp-pwr-en {
+			rockchip,pins = <1 24 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
 };
 
 &sdio0 {
@@ -50,6 +87,11 @@ &sound {
 	hp-det-gpio = <&gpio1 RK_PA0 GPIO_ACTIVE_HIGH>;
 };
 
+&tcphy0 {
+	extcon = <&virtual_pd>;
+	status = "okay";
+};
+
 &uart0 {
 	status = "okay";
 
-- 
2.38.0

